src/firestore/api-sort.ts

```diff
@@ -225,6 +230,19 @@ function compareArrayConfig(a?: API.ArrayConfig, b?: API.ArrayConfig): number {
   return ARRAY_CONFIG_SEQUENCE.indexOf(a) - ARRAY_CONFIG_SEQUENCE.indexOf(b);
 }
 
+function compareVectorConfig(a?: API.VectorConfig, b?: API.VectorConfig): number {
+  if (a === undefined) {
+    if (b === undefined) {
+      return 0;
+    } else {
+      return 1;
+    }
+  } else if (b === undefined) {
+    return -1;
```

Comment:
Is this ordering deliberate? If `a` has a higher `dimension` than `b`, then this function will have a positive return value. But if `b` is `undefined` and `a` is not (and thus presumably has non-zero degree), this function will have a negative return value.